modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -292,7 +284,7 @@ public void updateMessageID(String queueName, String nodeId, long startMessageId
             firstMessageId = startMessageIdInTheSlot;
         }
 
-	    if (slotRecoveryScheduled.compareAndSet(true, true)) {
```

Comment:
Yes. The important thing to note is we are not setting a new value here. Therefore get() == compareAndSet() (Just for this special case).